projects/onto-search/src/lib/onto-search-field/search/search.component.html

```diff
@@ -19,16 +20,17 @@
           [matChipInputFor]="selectionList"
           [matChipInputSeparatorKeyCodes]="separatorKeysCodes"
           (matChipInputTokenEnd)="add($event)"
-          (input)="onInputChange($event.target.value)">
+          (input)="onInputChange($event.target.value)"
+          appCypressData="search-field-input">
       </mat-chip-list>
       <mat-autocomplete #auto="matAutocomplete" (optionSelected)="selected($event)">
-        <mat-option class="state-option" *ngFor="let state of filteredStates | async" [value]="state">
-          <ng-container *ngTemplateOutlet="currentTemplate; context: {$implicit: state}">
+        <mat-option class="state-option" *ngFor="let state of filteredStates | async; let index = index" [value]="state">
+          <ng-container *ngTemplateOutlet="currentTemplate; context: {$implicit: {state: state, index: index}}">
```

Comment:
You don't need to have everything as implicit context.
`context: {$implicit: {state}, index}` is also valid and you will have `let-item; let-index`

Reply:
its a bit more readable to me

Reply:
Most people implementing a custom template won't care about the index of option.

Reply:
Either way, template should be documented.